pkg/migrations/coordinator_test.go

```diff
@@ -0,0 +1,163 @@
+	"github.com/xataio/pgroll/pkg/schema"
+)
+
+func TestCoordinator(t *testing.T) {
+	type testCase map[string]struct {
+		actions       []DBAction
+		expectedOrder []string
+	}
+
+	testCases := testCase{
+		"empty": {
+			actions:       []DBAction{},
+			expectedOrder: []string{},
+		},
+		"single action": {
+			actions: []DBAction{
+				NewRenameDuplicatedColumnAction(nil, &schema.Table{Name: "t1"}, "column1"),
+			},
+			expectedOrder: []string{"rename_duplicated_t1_column1"},
+		},
+		"multiple actions with duplicates": {
+			actions: []DBAction{
+				NewRenameDuplicatedColumnAction(nil, &schema.Table{Name: "t1"}, "column1"),
+				NewRenameDuplicatedColumnAction(nil, &schema.Table{Name: "t1"}, "column2"),
+				NewRenameDuplicatedColumnAction(nil, &schema.Table{Name: "t1"}, "column1"), // Duplicate
+			},
+			expectedOrder: []string{"rename_duplicated_t1_column2", "rename_duplicated_t1_column1"},
+		},
+		"multiple actions with mutiple duplicated for renaming": {
+			actions: []DBAction{
+				NewDropColumnAction(nil, "t1", "column1"),
+				NewRenameDuplicatedColumnAction(nil, &schema.Table{Name: "t1"}, "column1"),
+				NewDropColumnAction(nil, "t1", "column2"),
+				NewRenameDuplicatedColumnAction(nil, &schema.Table{Name: "t1"}, "column2"),
+				NewDropColumnAction(nil, "t1", "column3"),
+				NewRenameDuplicatedColumnAction(nil, &schema.Table{Name: "t1"}, "column3"),
+				NewDropColumnAction(nil, "t1", "column1"),
+				NewRenameDuplicatedColumnAction(nil, &schema.Table{Name: "t1"}, "column1"),
+				NewDropColumnAction(nil, "t1", "column2"),
+				NewRenameDuplicatedColumnAction(nil, &schema.Table{Name: "t1"}, "column2"),
+			},
+			expectedOrder: []string{
+				"drop_column_t1_column3",
+				"rename_duplicated_t1_column3",
+				"drop_column_t1_column1",
+				"rename_duplicated_t1_column1",
+				"drop_column_t1_column2",
+				"rename_duplicated_t1_column2",
+			},
+		},
+		"add same column multiple times to same column": {
+			actions: []DBAction{
+				NewCreateTableAction(nil, "test_table", "", ""),
+				NewAddColumnAction(nil, "t1", Column{Name: "column1"}, false),
+				NewAddColumnAction(nil, "t1", Column{Name: "column1"}, false), // Duplicate
+			},
+			expectedOrder: []string{"create_table_test_table", "add_column_t1_column1"},
+		},
+		"create table multiple time with different statement with the same name": {
+			actions: []DBAction{
+				NewCreateTableAction(nil, "test_table", "", ""),
+				NewCreateTableAction(nil, "test_table", "id INT", ""),
+				NewCommentTableAction(nil, "test_table", ptr("This is a test table")),
+				NewCommentColumnAction(nil, "test_table", "id", ptr("This is a test column")),
+			},
+			expectedOrder: []string{"create_table_test_table", "comment_table_test_table", "comment_column_test_table_id"},
+		},
+		"drop default value on column": {
+			actions: []DBAction{
+				NewCreateTableAction(nil, "test_table", "", ""),
+				NewAddColumnAction(nil, "column1", Column{Default: ptr("default_value")}, false),
```

Comment:
This action looks odd compared to how the others are used; the second argument is meant to be the table name, and the `Column` in the third argument should have a name.